src/main/java/reactor/core/publisher/FluxOnBackpressureBufferStrategy.java

```diff
@@ -108,6 +103,30 @@ public long getPrefetch() {
 		}
 
 		@Override
+		public Object scan(Attr key) {
+			switch (key) {
+				case PARENT:
+					return s;
+				case REQUESTED_FROM_DOWNSTREAM:
+					return requested;
+				case TERMINATED:
+					return done && isEmpty();
+				case CANCELLED:
+					return s == Operators.cancelledSubscription();
```

Comment:
incorrect, s is never set to cancelledSubscription by cancel(), `return cancelled;` instead